satpy/readers/mviri_l1b_fiduceo_nc.py

```diff
@@ -513,11 +514,19 @@ def get_time(self):
         """Get time coordinate.
 
         Variable is sometimes named "time" and sometimes "time_ir_wv".
+        FillValues in time are set to NaT.
         """
         try:
-            return self["time_ir_wv"]
+            time = self["time_ir_wv"]
         except KeyError:
-            return self["time"]
+            time = self["time"]
+
+        timeOffset = time.attrs["add_offset"]
+        condition = time == time.attrs["_FillValue"]
+
+        time = xr.where(condition, numpy.nan, time + timeOffset)
+        time = (time*1e9).astype("datetime64[ns]")
```

Comment:
how about
```suggestion
        time = time.astype("datetime64[s]").astype("datetime64[ns]")
```
this way you don't risk integer overflow

Reply:
see conversation above